test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/OpenIdConnectIdentityFlowTests.cs

```diff
@@ -0,0 +1,640 @@
+        {
+            options.StateDataFormat = new AuthenticationPropertiesFormater();
+            options.Notifications =
+                new OpenIdConnectAuthenticationNotifications
+                {
+                    SecurityTokenValidated = (notification) =>
+                    {
+                        notification.SkipToNextMiddleware();
+                        return Task.FromResult<object>(null);
+                    }
+                };
+        }
+
+        private static void DefaultOptions(OpenIdConnectAuthenticationOptions options)
+        {
+            options.StateDataFormat = new AuthenticationPropertiesFormater();
+        }
+
+        #endregion
+
+        private static TestServer CreateServer(Action<OpenIdConnectAuthenticationOptions> configureOptions, string scheme)
+        {
+            return TestServer.Create(app =>
+            {
+                app.UseOpenIdConnectAuthenticationPublic(configureOptions);
+                app.Use(async (context, next) =>
+                {
+                    await next();
+                });
+            },
+            services =>
+            {
+                services.AddDataProtection();
+            });
+        }
+    }
+
+    /// <summary>
+    /// Extension specifies <see cref="OpenIdConnectAuthenticationMiddlewarePublic"/> as the middleware.
+    /// </summary>
+    public static class OpenIdConnectAuthenticationExtensions
+    {
+        /// <summary>
+        /// Adds the <see cref="OpenIdConnectAuthenticationMiddleware"/> into the ASP.NET runtime.
+        /// </summary>
+        /// <param name="app">The application builder</param>
+        /// <param name="options">Options which control the processing of the OpenIdConnect protocol and token validation.</param>
+        /// <returns>The application builder</returns>
+        public static IApplicationBuilder UseOpenIdConnectAuthenticationPublic(this IApplicationBuilder app, Action<OpenIdConnectAuthenticationOptions> configureOptions)
+        {
+            return app.UseMiddleware<OpenIdConnectAuthenticationMiddlewarePublic>(new ConfigureOptions<OpenIdConnectAuthenticationOptions>(configureOptions));
+        }
+    }
+
+    public class OpenIdConnectAuthenticationContext : IAuthenticateContext
+    {
+        public OpenIdConnectAuthenticationContext(string scheme = null)
+        {
+            AuthenticationScheme = scheme ?? OpenIdConnectAuthenticationDefaults.AuthenticationScheme;
+        }
+
+        public string AuthenticationScheme
+        {
+            get;
+            set;
+        }
+
+        public void Authenticated(ClaimsPrincipal principal, IDictionary<string, string> properties, IDictionary<string, object> description)
+        {
+        }
+
+        public void NotAuthenticated()
+        {
+        }
+    }
+
+    public class OpenIdConnectAuthenticationHandlerPublic : OpenIdConnectAuthenticationHandler
```

Comment:
When I saw this I was thinking you meant to rename the entire file OpenIdConnectAuthenticationHandlerTests, which I think is more accurate. I prefer OpenIdConnectAuthenticaitonHandlerTest over TestOpenIdConnectAuthenticationHandler as the prefix makes finding things in sorted lists slower cause everything starts with T...